examples/max_trials_callback.py

```diff
@@ -0,0 +1,42 @@
+"""
+Optuna example that demonstrates setting maximum number of trials in
+a shared database when multiple workers are used.
+
+In this example, we optimize a simple quadratic function. We use multiple
+script runs(workers) to demonstrate the use of max_trial_callbacks,
+which essentially allows user to set maximum number of trials
+regardless of the number of workers/scripts running the Trials.
+
+"""
+
+import optuna
+from optuna.trial import TrialState
+
+
+max_trials = 10
+
+
+def max_trial_callback(study, trial):
+    # we consider all the running states and already completed states.
+    n_complete = len(
+        study.get_trials(deepcopy=False, states=[TrialState.COMPLETE, TrialState.RUNNING])
+    )
+    if n_complete >= max_trials:
+        study.stop()
+
+
+def objective(trial):
+    x = trial.suggest_uniform("x", 0, 10)
+    return x ** 2
+
+
+if __name__ == "__main__":
+    _ = optuna.create_study(
+        study_name="test",
+        storage="sqlite:///database.sqlite",
+    )

```

Comment:
`create_study` can load exising study with `load_if_exists=True`.

```suggestion
    study = optuna.create_study(
        study_name="test",
        storage="sqlite:///database.sqlite",
        load_if_exists=True
    )

```